axiom/nr_segment_message.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "nr_axiom.h"
+
+#include <stdio.h>
+
+#include "nr_header.h"
+#include "nr_segment_message.h"
+#include "nr_segment_private.h"
+#include "util_strings.h"
+#include "util_url.h"
+
+/*
+ * Purpose : Set all the typed message attributes on the segment.
+ */
+static void nr_segment_message_set_attrs(
+    nr_segment_t* segment,
+    const nr_segment_message_params_t* params,
+    nrtxnopt_t options) {
+  nr_segment_message_t message_attributes = {0};
+
+  message_attributes.message_action = params->message_action;
+
+  if (options.message_tracer_segment_parameters_enabled) {
+    message_attributes.destination_name = nr_strempty(params->destination_name)
```

Comment:
Could we have `nr_segment_set_message()` handle the `nr_strempty()` check (since it is already doing a `NULL` check) and have this code just pass the pointer through w/o additional conditionals here.

Reply:
Good idea!
Please see: [6aec35d](https://github.com/newrelic/newrelic-php-agent/pull/991/commits/6aec35d7650ec9941a75c8f8198a2f14765de0f3)